### PR TITLE
serviceaccount: move service account token functionality to go-jose

### PR DESCRIPTION
**Pull request by mikedanese**
This is a non-functional change that makes some of my later work easier.

### CODE REVIEW

File: pkg/serviceaccount/jwt.go [hunk 1]
```diff
@@ -122,129 +126,110 @@ type jwtTokenAuthenticator struct {
...
-			return nil, false, err
+	public := &jwt.Claims{}
+	private := &privateClaims{}
+
+	var (
+		found   bool
+		errlist []error
+	)
+	for _, key := range j.keys {
+		if err := tok.Claims(key, public); err != nil {
```
Comment (ericchiang) [inline pkg/serviceaccount/jwt.go]: I haven't used the `jwt` package too much, but it seems like this should be

```
if err := tok.Claims(key, public, private); err != nil {
    // ...
}
```

**Comment by jbeda** [comment]
Looks good at first glance but want to go through this carefully because security.  Ran out of time tonight so if others get to it before me go for it.

File: pkg/serviceaccount/jwt.go (hunk 1 above)
Comment (ericchiang) [reply pkg/serviceaccount/jwt.go]: It's also a little annoying that Claims doesn't let us enforce the `alg` value. E.g. if we ever started signing tokens with `RS512`, this code would still allow users to pass tokens claiming to be signed with `RS256`.

**Review by ericchiang — COMMENTED** [review]
couple nits

**Comment by ericchiang** [comment]
Also, thanks for not changing the unit tests (assuming they pass).

File: pkg/serviceaccount/jwt.go (hunk 1 above)
Comment (mikedanese) [reply pkg/serviceaccount/jwt.go]: Is that desirable? What does it solve? The algorithm is available on the tok struct if we want to do this.

Comment (cjcullen) [reply pkg/serviceaccount/jwt.go]: +1 to Eric's suggestion. Makes the code a little cleaner.

**Comment by cjcullen** [comment]
Looks good to me. I'll let @ericchiang have the final word.

**Pull request merged**